Rocket.Chat/API/Requests/General/PublicSettingsRequest.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  PublicSettingsRequest.swift
+//  Rocket.Chat
+//
+//  Created by Filipe Alvarenga on 02/04/18.
+//  Copyright © 2018 Rocket.Chat. All rights reserved.
+//
+
+import Foundation
+typealias PublicSettingsResult = APIResult<PublicSettingsRequest>
+
+class PublicSettingsRequest: APIRequest {
+    let requiredVersion = Version(0, 62, 2)
+    let path = "/api/v1/settings.public"
+    let query: String? =
+    """
+    fields={
+        "type":1
+    }&query={
+        "_id":{
+            "$in":[
+                "Site_Url",
+                "CDN_PREFIX",
+                "Site_Name",
+                "Assets_favicon_512",
+                "UI_Use_Real_Name",
+                "UI_Allow_room_names_with_special_chars",
+                "Favorite_Rooms",
+                "Accounts_OAuth_Google",
+                "Accounts_OAuth_Facebook",
+                "Accounts_OAuth_Github",
+                "Accounts_OAuth_Gitlab",
+                "Accounts_OAuth_Linkedin",
+                "LDAP_Enable",
+                "CAS_enabled",
+                "CAS_login_url",
+                "Accounts_ShowFormLogin",
+                "Accounts_RegistrationForm",
+                "Accounts_PasswordReset",
+                "Accounts_EmailOrUsernamePlaceholder",
+                "Accounts_PasswordPlaceholder",
+                "Accounts_EmailVerification",
+                "Accounts_AllowUserProfileChange",
+                "Accounts_AllowUserAvatarChange",
+                "Accounts_AllowRealNameChange",
+                "Accounts_AllowUsernameChange",
+                "Accounts_AllowEmailChange",
+                "Accounts_AllowPasswordChange",
+                "FileUpload_Storage_Type",
+                "Message_HideType_uj",
+                "Message_HideType_ul",
+                "Message_HideType_au",
+                "Message_HideType_mute_unmute",
+                "Message_HideType_ru",
+                "Message_ShowDeletedStatus",
+                "Message_AllowDeleting",
+                "Message_AllowDeleting_BlockDeleteInMinutes",
+                "Message_ShowEditedStatus",
+                "Message_AllowEditing",
+                "Message_AllowEditing_BlockEditInMinutes",
+                "Accounts_CustomFields"
```

Comment:
I think these would be more maintainable if they were in a separate array then you could interpolate them into the query.

```swift
    """
    fields={
        "type":1
    }&query={
        "_id":{
            "$in":[
                \(settings.map { "\"\($0)\"" }.joined(separator: ","))
            ]
        }
    }
"""
```

settings could be stored in a static property of AuthSettings then passed in the request's init

Reply:
@cardoso I think that it would still be the same pain to maintain as it already is if we split a huge hardcoded string into an array with a lot of hardcoded strings. 

The pros of your suggestion in my opinion:

- We at least split the fields from the URL, so we make the string smaller, but in different places. 
- It gives us the ability to easily exclude some fields from the array if we wind up needing to request just one setting.

The cons: 

- It adds a new property that in the current scenario will never be changed because we always request all the fields, so it would be just a constant array of hardcoded strings. 

In my opinion, we will add complexity to our code and gain almost nothing thinking about maintainability. I mean, we would not have the giant string in our code, but we would wind up adding more code only to break this string in a way that it would look better but honestly, I don't think it will be more maintainable because in some way we will need to have this huge string. 

If it was not a hardcoded string that will never change, but a URL built with dynamic parameters I would 100% agree with you. 

Besides respectfully disagreeing with your point about maintainability I have nothing against your suggestion, I'm just answering what I think :P 

Let me know if you still want me to implement it, so I will do it.